### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Vayts**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,32 @@
...
+      case 'clear':
+        Object.keys(copyState).forEach((key) => delete copyState[key]);
+        break;
+      case 'removeProperties':
+        item.keysToRemove.forEach((key) => {
+          delete copyState[key];
+        });
+        break;
+      default:
+        break;
```
Comment (tiserett) [inline src/transformStateWithClones.js:30]: I guess it's gonna be better, if you add gaps between blocks

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  actions.forEach((item) => {
+    const copyState = result.length
```
Comment (tiserett) [inline src/transformStateWithClones.js:13]: copyState sounds like an action
I guess stateCopy is a better name

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,32 @@
...
+      ? { ...result[result.length - 1] }
+      : { ...state };
```
Comment (tiserett) [inline src/transformStateWithClones.js:15]: to get last element it's easier to use at(-1)

btw, not sure if you need it here, you can just create this variable outside of loop, work with it, and every time you push it -> just push it's copy

Comment (Vayts) [reply src/transformStateWithClones.js]: This works on LTS node, but not on 14.21.3..... I don't understand why...

**Review by edgergard — APPROVED** [review]

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,35 @@
...
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete stateCopy[key];
+        });
+        break;
+
+      default:
+        break;
```
Comment (Moroz-Dmytro) [inline src/transformStateWithClones.js:33]: Would be great to throw an error in such case

File: src/transformStateWithClones.js [hunk 5]
```diff
@@ -7,7 +7,35 @@
...
+    }
+    stateHistory.push(stateCopy);
```
Comment (Moroz-Dmytro) [inline src/transformStateWithClones.js:35]:
```suggestion
    }
    
    stateHistory.push(stateCopy);
```

**Review by Moroz-Dmytro — CHANGES_REQUESTED** [review]
Well done 👍 
Small comments

**Review by roman-mirzoian — APPROVED** [review]
Well done 👍

**Pull request closed without merging**